### PR TITLE
feat(rework): reworked the package to make it platform agnostic

### PR DESCRIPTION
**Pull request by dominikfeininger**
see readme

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,20 +1,65 @@
...
+    ]
+  },
+  "husky": {
+    "hooks": {
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
+      "pre-commit": "lint-staged"
+    }
+  },
+  "devDependencies": {
+    "wdio-ui5": "0.0.4",
```
Comment (vobu) [inline package.json]: this causes a circular reference to wdio-ui5 itself :) -> please delete

File: test/ui5-app/controller/BaseController.js [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
+sap.ui.define(
+    ["sap/ui/core/mvc/Controller", "sap/ui/core/routing/History"],
     /**
      * @param {typeof sap.ui.core.mvc.Controller} Controller
      * @param {typeof sap.ui.core.routing.History} History
      * @yields {typeof sap.ui.core.mvc.Controller}
      */
-    (Controller, History) => {
-        "use strict";
+    function (Controller, History) {
```
[the remark this answers is not in the document]
Comment (dominikfeininger) [reply test/ui5-app/controller/BaseController.js:8]: Due to the lack of babel transpile I figured ES5 is better.

File: test/ui5-app/controller/Main.controller.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+sap.ui.define(["test/Sample/controller/BaseController", "sap/m/MessageToast"], function (Controller, MessageToast) {
```
[the remark this answers is not in the document]
Comment (dominikfeininger) [reply test/ui5-app/controller/Main.controller.js:1]: Due to the lack of babel transpile I figured ES5 is better.

File: test/wdio.conf.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+        // can this be done in config?
+        browser.url("index.html")
+
+        // then use the get* Calls
+        console.log("configurations: " + JSON.stringify(wdi5().getUtils().getConfig()))
+
+        // ui5 shim setup
+        wdi5().getWDioUi5().setup(browser) // use wdio hooks for setting up wdio<->ui5 bridge
+        wdi5().getWDioUi5().injectUI5(browser) // needed to let the instance know that UI5 is now available for work
```
[the remark this answers is not in the document]
Comment (dominikfeininger) [reply test/wdio.conf.js:197]: Added "TODO: clarify why this is needed and what about `wdi5().getWDioUi5().setup(browser)`".

Not needed for other platforms like appium tests. Some guess is: "browser opens multiple windows, this causes issues if `before` does not run again.

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# editorconfig.org
```
[the remark this answers is not in the document]
Comment (dominikfeininger) [reply .editorconfig:1]: tried to add globals like "sap", "browser", "driver" etc. but wasn't working anyways yet

[the remark this answers is not in the document]
Comment (jkoenig134) [reply .editorconfig]: globals can be added in the tsconfig.

File: package.json [hunk 6]
```diff
@@ -1,20 +1,64 @@
...
-    "dev": "cd packages/ui5-app && yarn dev",
-    "test": "cd packages/ui5-app && wdio wdio.conf.js",
-    "test:watch": "cd packages/ui5-app && wdio wdio.conf.js --watch"
+    "_startApp": "http-server ./test/ui5-app --p 8888 --proxy https://services.odata.org",
+    "_test": "wdio ./test/wdio.conf.js",
+    "test": "run-p _startApp _test"
+  },
+  "screenshotPath": "./test/report/screenshots",
+  "publishConfig": {
+    "registry": "https://nexus.js-soft.com/repository/npm-private/"
```
Comment (jkoenig134) [inline package.json:14]: Remove publish config

File: package.json [hunk 7]
```diff
@@ -1,20 +1,64 @@
...
-  "license": "Apache-2.0",
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/petermuessig/ui5-ecosystem-showcase.git"
-  },
+  "name": "wdio-ui5",
+  "version": "0.0.5",
+  "description": "package to enhance appium driven wdio test for cordova wrapped ui5 applications",
+  "main": "index.js",
+  "repository": "git@bitbucket.org:dominikfeininger/testing-wdi5.git",
```
Comment (jkoenig134) [inline package.json:6]: Set to correct git repository and use npm-style repository format

**Pull request merged**